alf/algorithms/trac_algorithm.py

```diff
@@ -59,44 +60,33 @@ class TracAlgorithm(ActorCriticAlgorithm):
 
     def __init__(self,
                  action_spec,
-                 actor_network: DistributionNetwork,
-                 value_network: Network,
+                 ac_algorithm_cls=ActorCriticAlgorithm,
                  action_dist_clip_per_dim=0.01,
-                 loss=None,
-                 loss_class=ActorCriticLoss,
-                 optimizer=None,
                  debug_summaries=False,
                  name="TracAlgorithm"):
         """Create an instance TracAlgorithm.
 
         Args:
             action_spec (nested BoundedTensorSpec): representing the actions.
-            action_dist_clip_per_dim (float): adjust step if the average
-                distance between the old and new action distributions exceeds
-                this value multiplied with dimension.
-            actor_network (DistributionNetwork): A network that returns nested
-                tensor of action distribution for each observation given observation
-                and network state.
-            value_network (Network): A function that returns value tensor from neural
-                net predictions for each observation given observation and nwtwork
-                state.
-            loss (None|ActorCriticLoss): an object for calculating loss. If
-                None, a default loss of class loss_class will be used.
-            loss_class (type): the class of the loss. The signature of its
-                constructor: loss_class(action_spec, debug_summaries)
-            optimizer (tf.optimizers.Optimizer): The optimizer for training
+            ac_algorithm_cls (type): Actor Critic Algorithm cls (AC, PPO, SAC...)
```

Comment:
DDPG can also be supported. We only need to support the distance between two deterministic action distribution.

Reply:
added

Reply:
`kl_divergence` implemented in `SquashToSpecNormal` checks action mean and magnitude for two distributions, and it fails in graph mode,  have not found the reason yet.